### PR TITLE
[webview_flutter] Added Android implementation of PlatformWebViewWidget

### PR DESCRIPTION
**Pull request by mvanbeusekom**
Adds the Android implementation of the `PlatformWebViewWidget` to the v4 Android implementation.

Part of issue https://github.com/flutter/flutter/issues/94051

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_webview_controller.dart [hunk 1]
```diff
@@ -320,3 +320,70 @@ class AndroidJavaScriptChannelParams extends JavaScriptChannelParams {
...
+              : AndroidWebViewWidgetCreationParams
+                  .fromPlatformWebViewWidgetCreationParams(params),
+        );
+
+  AndroidWebViewWidgetCreationParams get _androidParams =>
+      params as AndroidWebViewWidgetCreationParams;
+
+  @override
+  Widget build(BuildContext context) {
+    return AndroidView(
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_webview_controller.dart:379]: I think this still uses VirtualDisplay. This should use the TLHC implementation with `PlatformViewLink` and `PlatformViewsService.initSurfaceAndroidView`:
https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_android/lib/webview_surface_android.dart#L47

**Review by bparrishMines — CHANGES_REQUESTED** [review]

**Review by bparrishMines — APPROVED** [review]
LGTM with a nit. 

It looks like the analyze check is still failing. It took me a few tries to figure it out since you have to make it work with the current Flutter version and the 2 previous stable versions. 

This combination might work:

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';
import 'package:flutter/services.dart';
```

And remove `foundation` and `gestures`.

File: packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_webview_controller.dart [hunk 2]
```diff
@@ -320,3 +322,86 @@ class AndroidJavaScriptChannelParams extends JavaScriptChannelParams {
...
+              : AndroidWebViewWidgetCreationParams
+                  .fromPlatformWebViewWidgetCreationParams(params),
+        );
+
+  AndroidWebViewWidgetCreationParams get _androidParams =>
+      params as AndroidWebViewWidgetCreationParams;
+
+  @override
+  Widget build(BuildContext context) {
+    return PlatformViewLink(
```
[the remark this answers is not in the document]
Comment (mvanbeusekom) [reply packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_webview_controller.dart:389]: Updated

File: packages/webview_flutter/webview_flutter_android/pubspec.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ publish_to: none
 
 environment:
   sdk: ">=2.17.0 <3.0.0"
-  flutter: ">=3.0.0"
+  flutter: ">=3.3.0"
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_android/pubspec.yaml:10]: I debugged the error from the legacy analyze test and it looks like this is the culprit. You will also need to update the minimum Flutter version of `webview_flutter` to `3.3.0`.

**Pull request merged**